p2p/src/network_builder.rs

```diff
@@ -99,7 +99,6 @@ impl<REQ: RpcRequest, RESP: RpcResponse> NetworkBuilder<REQ, RESP> {
         }))
         .build();
 
-        swarm.listen_on(self.listen_address)?;
```

Comment:
Previous to all the work done in this PR, this would return almost immediately and queue the listening, and eventually it will happen. However, this means that when `build()` returns, its possible we still haven't started listening, and don't know our local address.